.github/scripts/decrypt_test_key.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+# Decrypt the file
+mkdir $HOME/secrets
+# --batch to prevent interactive command --yes to assume "yes" for questions
+gpg --quiet --batch --yes --decrypt --passphrase="$TEST_KEY_PASSPHRASE" \
+--output config/credentials/test.key config/credentials/test_key.gpg
```

Comment:
Instead of encrypting&decrypting test key manually, it's a better idea to add it to repo settings, as a repository secret (settings > secrets and variables > actions > repository secrets).
And then add to the workflow in the step when you're running tests:
```
        env:
          RAILS_ENV: test
          RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}
```
RAILS_MASTER_KEY value for github actions should be test.key value.
And then you don't need this script at all. Repository secrets are already encrypted, so there's no need to do it manually.